### DISCUSSION TITLE
Admin Settings persistence on error

### DISCUSSION THREAD

**Discussion by moritztim** [dani-garcia/vaultwarden#3811] [Ideas]
It would be nice not to lose all changes made to settings in the admin panel if it turns out after saving that you've made a mistake. The settings should persist client side, ideally even with inline errors, so you can update them accordingly.

**Comment by BlackDex** [comment]
Just delete the `config.json` in your data folder.

**Reply by moritztim** [reply]
Sorry I was a little imprecise, I've updated the description to better explain what I meant.

**Reply by BlackDex** [reply]
If there is an error, in the sense that the server doesn't allow the setting, it will just not save it if I'm correct. If it does, please let us know which setting it was.

**Reply by moritztim** [reply]
That's correct, it doesn't save. The issue is that if you change a whole bunch of things, none of them are saved. That's why it would be nice to have them filled out client side as they where before attempting to save

**Reply by BlackDex** [reply]
That might be possible.
